examples/bidspath.py

```diff
@@ -0,0 +1,159 @@
+print(bids_path_new.subject)
+
+# Option 2: Update the existing BIDSPath in-place.
+bids_path.update(subject=subject)
+print(bids_path.subject)
+
+###############################################################################
+# In this example, we are going to update the existing ``BIDSPath`` using its
+# ``update()`` method. But note that all parameters we pass to this method can
+# also be used when creating a ``BIDSPath``.
+#
+# Many studies consist of multiple **sessions**. As you may have guessed,
+# BIDS specifies how to store data for each session, and consequently,
+# ``BIDSPath`` handles this for you too! Let's update our ``BIDSPath`` with
+# information on our experimental session, and try to retrieve it again via
+# ``.session``.
+
+session = 'test'
+bids_path.update(session=session)
+print(bids_path.session)
+
+###############################################################################
+# Now that was easy! We're almost there! We also need to specify a
+# **data type**, i.e., ``meg`` for MEG data, ``eeg`` and ``ieeg`` for EEG and
+# iEEG data, or ``anat`` for anatomical MRI scans. Typically, MNE-BIDS will
+# infer the data type of your data automatically, for example when writing data
+# using `mne_bids.write_raw_bids`. For the sake of this example, however, we
+# are going to specify the data type explicitly.
+
+datatype = 'meg'
+bids_path.update(datatype=datatype)
+print(bids_path.datatype)
+
+###############################################################################
+# Excellent! Let's have a look at the path we have constructed!
+print(bids_path)
+
+###############################################################################
+# As you can see, ``BIDSPath`` automatically arranged all the information we
+# provided such that it creates a valid BIDS folder structure. You can also
+# retrieve a `pathlib.Path` object of this path:
+
+pathlib_path = bids_path.fpath
+pathlib_path
+
+###############################################################################
+# Let's have a closer look at the components of our ``BIDSPath`` again.
+
+bids_path
+
+###############################################################################
+# The most interesting thing here is probably the **basename**. It's what
+# MNE-BIDS uses to name individual files. The basename consists of a set of
+# so-called **entities**, which are concatenated using underscores. You can
+# access it directly:
+
+bids_path.basename
+
+###############################################################################
+# The two entities you can see here are the ``subject`` entity (``sub``) and
+# the ``session`` entity (``ses``). Each entity name also has a value; for
+# ``sub``, this is ``123``, and for ``ses``, it is ``test`` in our example.
+# Entity names (or "keys") and values are separated via hyphens.
+# BIDS knows a much larger number of entities, and MNE-BIDS allows you to make
+# use of them. To get a list of all supported entities, use:
```

Comment:
```suggestion
# use of them. To get a list of all entities in bids_path, use:
```

I presume you don't get a list of all entities supported by BIDS?

Reply:
You get a list of all entities supported in MNE-BIDS

Reply:
@jasmainak 

```python
In [25]: p = BIDSPath(root='/tmp/foo', subject='01', session='test',
    ...:              task='dostuff', datatype='meg', suffix='meg',
    ...:              extension='.fif')

In [26]: p.entities
Out[26]:
OrderedDict([('subject', '01'),
             ('session', 'test'),
             ('task', 'dostuff'),
             ('acquisition', None),
             ('run', None),
             ('processing', None),
             ('space', None),
             ('recording', None),
             ('split', None)])
```